perses/tests/test_geometry_engine.py

```diff
@@ -90,11 +108,220 @@ def test_run_geometry_engine():
 
     new_positions, logp_proposal = geometry_engine.propose(sm_top_proposal)
+    app.PDBFile.writeFile(top2, new_positions, file=test_pdb_file)
+    test_pdb_file.close()
     context.setPositions(new_positions)
     state2 = context.getState(getEnergy=True)
     print("Energy after proposal is: %s" %str(state2.getPotentialEnergy()))
 
+def test_existing_coordinates():
+    """
+    predUS
+    for each torsion, calculate position of atom1
+    """
+    molecule_name_2 = 'butane'
+    molecule2 = generate_initial_molecule(molecule_name_2)
+    sys, pos, top = oemol_to_openmm_system(molecule2, molecule_name_2)
+    import perses.rjmc.geometry as geometry
+    geometry_engine = geometry.FFAllAngleGeometryEngine({'test': 'true'})
+    structure = parmed.openmm.load_topology(top, sys)
+    torsions = [torsion for torsion in structure.dihedrals if not torsion.improper]
+    for torsion in torsions:
+        atom1_position = pos[torsion.atom1.idx]
+        atom2_position = pos[torsion.atom2.idx]
+        atom3_position = pos[torsion.atom3.idx]
+        atom4_position = pos[torsion.atom4.idx]
+        _internal_coordinates, _ = geometry_engine._cartesian_to_internal(atom1_position, atom2_position, atom3_position, atom4_position)
+        internal_coordinates = internal_in_units(_internal_coordinates)
+        recalculated_atom1_position, _ = geometry_engine._internal_to_cartesian(atom2_position, atom3_position, atom4_position, internal_coordinates[0], internal_coordinates[1], internal_coordinates[2])
+        n = np.linalg.norm(atom1_position-recalculated_atom1_position)
+        print(n)
+
+def internal_in_units(internal_coords):
+    r = internal_coords[0]*units.nanometers if type(internal_coords[0]) != units.Quantity else internal_coords[0]
+    theta = internal_coords[1]*units.radians if type(internal_coords[1]) != units.Quantity else internal_coords[1]
+    phi = internal_coords[2]*units.radians if type(internal_coords[2]) != units.Quantity else internal_coords[2]
+    return [r, theta, phi]
+
+def test_coordinate_conversion():
+    import perses.rjmc.geometry as geometry
+    geometry_engine = geometry.FFAllAngleGeometryEngine({'test': 'true'})
+    example_coordinates = units.Quantity(np.random.normal(size=[100,3]), unit=units.nanometers)
+    #try to transform random coordinates to and from
+    for i in range(200):
+        indices = np.random.randint(100, size=4)
+        atom_position = units.Quantity(np.array([ 0.80557722 ,-1.10424644 ,-1.08578826]), unit=units.nanometers)
+        bond_position = units.Quantity(np.array([ 0.0765,  0.1  ,  -0.4005]), unit=units.nanometers)
+        angle_position = units.Quantity(np.array([ 0.0829 , 0.0952 ,-0.2479]) ,unit=units.nanometers)
+        torsion_position = units.Quantity(np.array([-0.057 ,  0.0951 ,-0.1863] ) ,unit=units.nanometers)
+        rtp, detJ = geometry_engine._cartesian_to_internal(atom_position, bond_position, angle_position, torsion_position)
```

Comment:
Your documentation for `_cartesian_to_internal` says:
```
 Returns without units!
```
but then you feed the unitless `r` into `_internal_to_cartesian` which immediately removes units from `r`, assuming it is unit-bearing.

We will need some way to be more consistent with units in a refactoring PR.